substrate-processor/src/batch.ts

```diff
@@ -191,7 +220,14 @@ function mergeMaps<T>(a: Record<string, T>, b: Record<string, T>, mergeItems: (a
     return result
 }
 
-
+/**
+ * Given a list of {@link Range}, it counts the total number of blocks, adding each range until the chain height is 
```

Comment:
It counts the total number of blocks spanned by the ranges, up to the chain height.